### DISCUSSION TITLE
Show dialog in setup method on MacOS

### DISCUSSION THREAD

**Discussion by cbinzer** [tauri-apps/tauri#8190] [Q&A]
Hi,

I’am really new to Tauri and I’am trying to display a dialog on MacOS that is created in the setup method of the Tauri builder. So I have the following code:

```Rust
fn main() {
    tauri::Builder::default()
        .setup(|app| {
            let main_window = app.get_window("main").unwrap();
            dialog::blocking::message(Some(&main_window), "Hello", "Welcome back!");
            Ok(())
        })
        .run(tauri::generate_context!())
        .expect("error while running tauri application");
}
```

When I’am starting my app then I get the following error: `You are running RFD in NonWindowed environment, it is impossible to spawn dialog from thread different than main in this env.`

What I have todo to display the dialog in the setup mehtod on MacOS?

Thanks in advance

**Comment by FabianLars** [comment]
I feel like this is the same as https://github.com/tauri-apps/tauri/issues/6301 - maybe a workaround would be to use <https://docs.rs/rfd/latest/rfd/> directly? 🤔 (Edit: Only in the setup hook, later in tauri commands for example it needs tauri's wrapper api to handle the threading correctly)

**Reply by cbinzer** [reply]
Hey, thank you for the hint. This helped me. I adapted my code as follows:
```rust
fn main() {
    tauri::Builder::default()
        .setup(|app| {
            std::thread::spawn(move || {
                dialog::blocking::message(None::<&Window>, "Hello", "Welcome back!");
            });
            Ok(())
        })
        .run(tauri::generate_context!())
        .expect("error while running tauri application");
}
```
